src/libcore/num/f64.rs

```diff
@@ -17,217 +14,131 @@ use crate::mem;
-/// // deprecated way
-/// let min = std::f64::MIN_EXP;
-///
-/// // intended way
-/// let min = f64::MIN_EXP;
-/// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(
+    since = "1.46.0",
+    reason = "The associated constant `f64::MIN_EXP` is now prefered",
+    suggestion = "f64::MIN_EXP",
+)]
 pub const MIN_EXP: i32 = f64::MIN_EXP;
 
 /// Maximum possible power of 2 exponent.
-/// Use [`f64::MAX_EXP`](../../std/primitive.f64.html#associatedconstant.MAX_EXP) instead.
-///
-/// # Examples
-///
-/// ```rust
-/// // deprecated way
-/// let max = std::f64::MAX_EXP;
-///
-/// // intended way
-/// let max = f64::MAX_EXP;
-/// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(
+    since = "1.46.0",
+    reason = "The associated constant `f64::MAX_EXP` is now prefered",
+    suggestion = "f64::MAX_EXP",
+)]
 pub const MAX_EXP: i32 = f64::MAX_EXP;
 
 /// Minimum possible normal power of 10 exponent.
-/// Use [`f64::MIN_10_EXP`](../../std/primitive.f64.html#associatedconstant.MIN_10_EXP) instead.
-///
-/// # Examples
-///
-/// ```rust
-/// // deprecated way
-/// let min = std::f64::MIN_10_EXP;
-///
-/// // intended way
-/// let min = f64::MIN_10_EXP;
-/// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(
+    since = "1.46.0",
+    reason = "The associated constant `f64::MIN_10_EXP` is now prefered",
+    suggestion = "f64::MIN_10_EXP",
+)]
 pub const MIN_10_EXP: i32 = f64::MIN_10_EXP;
 
 /// Maximum possible power of 10 exponent.
-/// Use [`f64::MAX_10_EXP`](../../std/primitive.f64.html#associatedconstant.MAX_10_EXP) instead.
-///
-/// # Examples
-///
-/// ```rust
-/// // deprecated way
-/// let max = std::f64::MAX_10_EXP;
-///
-/// // intended way
-/// let max = f64::MAX_10_EXP;
-/// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(
+    since = "1.46.0",
+    reason = "The associated constant `f64::MAX_10_EXP` is now prefered",
+    suggestion = "f64::MAX_10_EXP",
+)]
 pub const MAX_10_EXP: i32 = f64::MAX_10_EXP;
 
 /// Not a Number (NaN).
-/// Use [`f64::NAN`](../../std/primitive.f64.html#associatedconstant.NAN) instead.
-///
-/// # Examples
-///
-/// ```rust
-/// // deprecated way
-/// let nan = std::f64::NAN;
-///
-/// // intended way
-/// let nan = f64::NAN;
-/// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(
+    since = "1.46.0",
+    reason = "The associated constant `f64::NAN` is now prefered",
+    suggestion = "f64::NAN",
+)]
 pub const NAN: f64 = f64::NAN;
 
 /// Infinity (∞).
-/// Use [`f64::INFINITY`](../../std/primitive.f64.html#associatedconstant.INFINITY) instead.
-///
-/// # Examples
-///
-/// ```rust
-/// // deprecated way
-/// let inf = std::f64::INFINITY;
-///
-/// // intended way
-/// let inf = f64::INFINITY;
-/// ```
 #[stable(feature = "rust1", since = "1.0.0")]
+#[rustc_deprecated(
+    since = "1.46.0",
+    reason = "The associated constant `f64::INFINITY` is now prefered",
+    suggestion = "f64::INFINITY",
```

Comment:
I am afraid that rustfix cannot handle this suggestion.

For example, what does rustfix do in this case?
```rust
use std::f64;
f64::INFINITY;
```

Reply:
I ran the snippet locally. rustfix doesn't do anything except for this case:
```diff
 #![warn(deprecated_in_future)]
 
 pub fn foo() -> f64 {
-    use std::f64 as float;
-    float::INFINITY
+    
+    f64::INFINITY
 }
```